applets/005-netperf/applet.py

```diff
@@ -81,17 +48,18 @@ def main(x, conf, args):
     arg_d["flow_offset"] = args[6].split(":")[1]
     arg_d["netserver_port"] = args[7].split(":")[1]
     # retrieve: source ip, source user name, destination ip, destination user name
-    arg_d["ip_source"] = conf['boxes'][arg_d["name_source"]]["interfaces"][0]['ip-address']
-    arg_d["user_source"] = conf['boxes'][arg_d["name_source"]]['user']
-    arg_d["ip_dest"] = conf['boxes'][arg_d["name_dest"]]["interfaces"][0]['ip-address']
-    arg_d["user_dest"] = conf['boxes'][arg_d["name_dest"]]['user']
+    arg_d["ip_source"] = conf.get_ip(arg_d["name_source"], 0)
+    arg_d["user_source"] = conf.get_user(arg_d["name_source"])
+    arg_d["ip_dest"] = conf.get_ip(arg_d["name_dest"], 0)
+    arg_d["user_dest"] = conf.get_user(arg_d["name_dest"])
 
     # potentially start parallel netperf instances
     # note: at this point, the distribution of the applet will be done, which
     # is probably not the best way to do (can cause unwanted time offsets)
     x.p.msg("Starting netperf applet on host {}\n".format(
                                                         arg_d["user_source"]))
-    contr_thread = Thread(target=controller_thread, args=(arg_d, ))
+    contr_thread = Thread(target=controller_thread, args=(x, arg_d, ))
```

Comment:
The idea at this point is to have non-dependant threads (no joining). This is due to the architecture of nas. Ressources are freed (as far as i understand), as soon as the threads finish. A netperf thread finishes, and thus frees ressources, directly after distribution of the netperf controller and it's startup.